### PR TITLE
Added the function `remove_unordered` to the LocalVector container.

### PR DESCRIPTION
**Pull request by AndreaCatania**
Added function `remove_unordered` to `LocalVector`.

Sometimes you don't need to keep the vector's order, so you can just remove the item by swapping the item you need to remove with the last one. With `remove_unordered` it's possible to avoid the extra copies that the `remove` function does to maintain the original insertion order.

### CODE REVIEW

**Comment by lawnjelly** [comment]
Totally agree to adding this, just a couple of small things:

* I'm wondering whether `remove_fast` is the best name? I don't really have an ideal name, but also think worth considering is:
`remove_unsorted` (or `remove_unordered`) to emphasise the difference from a standard remove?

This could be easier when reading in terms of finding bugs, because you can instantly mentally check whether these conditions are being violated, and also useful for newbies to understand what an existing bit of code is doing. `remove_fast` tells us very little, and could even lead to misguided attempts to replace `remove` with `remove_fast`.

On the flip side the alternative name doesn't emphasise that it is faster, this would have to be mentioned in the documentation / comments.

* For simple pod types especially, could it be better to omit the if check before doing the swap? Worst case, assigning a simple small type to itself costs very little. Whereas doing the if check every time in a tight loop could be expensive. Being a template you could also have a compile time switch for this depending on the sizeof the structure etc.

**Comment by AndreaCatania** [comment]
I agree and I think that `remove_unordered` is the best solution.

> For simple pod types especially, could it be better to omit the if check before doing the swap? Worst case, assigning a simple small type to itself costs very little. Whereas doing the if check every time in a tight loop could be expensive. Being a template you could also have a compile time switch for this depending on the sizeof the structure etc.

I don't think it would make any difference. The code into the `IF` is so little that the entire statement is likely to enter into the CPU cache, so the produced jump is so little that is not going to improve or change performances. Also, is super easy for the compiler understand the pattern and so cache the call statements in the correct way.

**Comment by AndreaCatania** [comment]
Renamed

File: core/local_vector.h [hunk 1]
```diff
@@ -82,6 +82,19 @@ class LocalVector {
 		}
 	}
 
+	/// Removes the item coping the last value into the position of the one to
```
Comment (Xrayez) [inline core/local_vector.h:85]: Typo? 😮

```suggestion
	/// Removes the item copying the last value into the position of the one to
```

**Comment by Xrayez** [comment]
Note that unit tests can also be written to cover the new functionality, although there's no tests written for `LocalVector` in the first place. 😕

See [Unit testing](https://docs.godotengine.org/en/latest/development/cpp/unit_testing.html#unit-testing) documentation for instructions, if you wish.

It's also possible to write stress tests, to demonstrate the efficiency of the new method when comparing with the existing `remove` (when running tests with `--test --duration`).

**Comment by AndreaCatania** [comment]
This is a good idea, to be honest. I'll add it yes.

**Comment by AndreaCatania** [comment]
@Xrayez I've added the unit `LocalVector` test, let me know.

**Review by Xrayez — APPROVED** [review]
Tests look good!

Need to fix formatting: https://github.com/godotengine/godot/pull/42270/checks?check_run_id=1681113902#step:5:15

**Comment by AndreaCatania** [comment]
Done

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**